libbeat/processors/add_kubernetes_metadata/indexers.go

```diff
@@ -39,15 +40,11 @@ type Indexers struct {
 	indexers []Indexer
 }
 
-//GenMeta takes in pods to generate metadata for them
-type GenMeta interface {
-	//GenerateMetaData generates metadata by taking in a pod as an input
-	GenerateMetaData(pod *Pod) common.MapStr
-}
-
-type IndexerConstructor func(config common.Config, genMeta GenMeta) (Indexer, error)
+// IndexerConstructor builds a new indexer from its settings
+type IndexerConstructor func(config common.Config, metaGen kubernetes.MetaGenerator) (Indexer, error)
 
-func NewIndexers(configs PluginConfig, metaGen *GenDefaultMeta) *Indexers {
+// NewIndexers  builds indexers object
+func NewIndexers(configs PluginConfig, metaGen kubernetes.MetaGenerator) *Indexers {
```

Comment:
should indexers and matchers be part of `add_kubernetes_metadata` or should they go into `kubernetes`?

Reply:
Metadata generation must be common, as we use the same for processor and autodiscover, but indexers/matchers are something used only from the processor, I would keep them there until we see the need to make them common